ledger-service/http-json/src/main/scala/com/digitalasset/http/json/JsonProtocol.scala

```diff
@@ -215,8 +215,41 @@ object JsonProtocol extends DefaultJsonProtocol {
   implicit val CreateCommandFormat: RootJsonFormat[domain.CreateCommand[JsObject]] = jsonFormat3(
     domain.CreateCommand[JsObject])
 
-  implicit val ExerciseCommandFormat: RootJsonFormat[domain.ExerciseCommand[JsValue]] =
-    jsonFormat5(domain.ExerciseCommand[JsValue])
+  implicit val ExerciseCommandFormat
+    : RootJsonFormat[domain.ExerciseCommand[JsValue, domain.ContractLocator[JsValue]]] =
+    new RootJsonFormat[domain.ExerciseCommand[JsValue, domain.ContractLocator[JsValue]]] {
+      @SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements"))
+      override def write(
+          obj: domain.ExerciseCommand[JsValue, domain.ContractLocator[JsValue]]): JsValue = {
+
+        val reference: JsObject =
+          ContractLocatorFormat.write(obj.reference).asJsObject("reference must be an object")
+
+        val fields = new collection.mutable.ListBuffer[(String, JsValue)]
+        fields.sizeHint(5)
+        fields ++= reference.fields.toList
+        fields += "choice" -> obj.choice.toJson
+        fields += "argument" -> obj.argument.toJson
+        if (obj.meta.isDefined)
+          fields += "meta" -> obj.meta.toJson
```

Comment:
There's no value here from not building fields in the functional style; use `++`.